sdk/go/common/resource/plugin/provider_plugin.go

```diff
@@ -290,8 +345,8 @@ func (p *provider) CheckConfig(urn resource.URN, olds,
 	molds, err := MarshalProperties(olds, MarshalOptions{
 		Label:         fmt.Sprintf("%s.olds", label),
 		KeepUnknowns:  allowUnknowns,
-		KeepSecrets:   p.acceptSecrets,
-		KeepResources: p.acceptResources,
+		KeepSecrets:   true,
```

Comment:
These should default to false. At the point of calling CheckConfig and DiffConfig we don't know if the provider supports secrets and resource refs and so can't send them. You could just remove setting them from these locations, there are other Keep settings (like KeepOutputValues) that we're not setting here, and the unset false value is correct for all of them.

Reply:
Note that all the "trues" for marshalling result back to the engine should stay true, we know the engine supports Secrets, its just passing values in to the provider that need to be conservative.

Reply:
🤦

Thanks for catching this.
I intended false, I thought false, and somehow I wrote true.

> I've filled in default values (false) for the configuration parameters
that they consumed before the provider was configured
because that's what they were using anyway.

Reply:
> You could just remove setting them from these locations, there are other Keep settings (like KeepOutputValues) that we're not setting here, and the unset false value is correct for all of them.

I missed this. Yeah, that's my preference; I leaned explicit here because I needed someone to confirm that this was okay to set to false. I'll drop those lines now.